File: app/src/main/java/com/antonchaynikov/triptracker/mainscreen/TripComponent.java
package com.antonchaynikov.triptracker.mainscreen;

import com.antonchaynikov.core.data.tripmanager.TripManagerModule;
import com.antonchaynikov.core.viewmodel.CommonViewModelModule;
import com.antonchaynikov.core.authentication.AuthModule;
import com.antonchaynikov.triptracker.injection.AppComponent;

import dagger.Component;

@TripScope
@Component(
        dependencies = AppComponent.class,
        modules = {TripModule.class, CommonViewModelModule.class, AuthModule.class, TripManagerModule.class})
public interface TripComponent {
    void inject(TripFragment fragment);
}

File: app/src/main/java/com/antonchaynikov/triptracker/TripComponent.kt
package com.antonchaynikov.triptracker

import com.antonchaynikov.tripscreen.TripFragment
import com.antonchaynikov.tripscreen.TripViewModel
import com.antonchaynikov.tripscreen.TripViewModelModule
import com.antonchaynikov.tripscreen.TripsScreenScope
import com.antonchaynikov.triptracker.application.AppComponent
import dagger.BindsInstance
import dagger.Component
import dagger.Subcomponent

@TripsScreenScope
@Subcomponent(modules = [TripViewModelModule::class])
interface TripComponent {
    fun inject(tripFragment: TripFragment)

    @Subcomponent.Builder
    interface Builder {
        fun build(): TripComponent
        @BindsInstance
        fun fragment(fragment: TripFragment): Builder
    }
}